data_science_pipeline/peerscout/api/main.py

```diff
@@ -10,16 +10,35 @@
 
 LOGGER = logging.getLogger(__name__)
 
-output = {
-    "reviewing_editor_recommendation": {
-        "person_ids": [],
-        "recommendation_html": "The recommended editors are ..."
-    },
-    "senior_editor_recommendation": {
-        "person_ids": [],
-        "recommendation_html": "The recommended editors are ..."
-    }
-}
+REQUEST_JSON_SCHEMA_PATH = os.path.join(os.path.dirname(__file__), 'input-json-schema.json')
+
+
+def response_json(
+    abstract: str,
+    reviewing_editors: list,
+    senior_editors: list) -> dict:
+
+    data = {}
+
+    if abstract == "":
+        data['reviewing_editor_recommendation']={}
+        data['senior_editor_recommendation']={}
+        data['reviewing_editor_recommendation']['person_ids'] = []
+        data['senior_editor_recommendation']['person_ids'] = []
+        data['reviewing_editor_recommendation']['recommendation_html'] = "The recommended editors are ..."
```

Comment:
here we could add the text that we don't have a recommendation. although I would make it based on the list of person ids. I was thinking that when the abstract is empty, we pass in an empty list of reviewing editors and senior editors to the function, but we could also do that at the start if you prefer.

Then something like (you can combine the two recommendations as they follow the same structure - just an idea):

```python
def get_recommendation_html(person_ids: List[str]) -> str:
    if not person_ids:
        return '<b>No</b> recommendation available'
    return '<b>The recommended editors are:</b> %s' % person_ids


def get_recommendation_json(person_ids: List[str]) -> dict:
    return {
       'person_ids': person_ids,
       'recommendation_html': get_recommendation_html(person_ids)
    }

def get_response_json(senior_editor_person_ids: List[str], reviewing_editor_person_ids: List[str]) -> dict:
    return {
        'senior_editor_recommendation': get_recommendation_json(senior_editor_person_ids),
        'reviewing_editor_recommendation': get_recommendation_json(reviewing_editor_person_ids)
    }
}

# ...
get_response_json(
    senior_editor_person_ids=suggested_senior_editor_person_ids if abstract else [],
    reviewing_editor_person_ids=suggested_reviewing_editor_person_ids if abstract else []
)
```

The above is just a suggestion. The code will likely change with the real recommendation anyway.


(I wouldn't make the function itself dump the JSON to str. I would usually leave that to the top most layer, which could for example add extra information or decide how it should be serialized)

In our case the advantage of using jsonify is, that it should also set the response content type.